File: day2/day2.py
def process_password(line):
    line = line.strip()
    parts = line.split(" ")
    minmax = parts[0].split("-")
    themin = int(minmax[0])
    themax = int(minmax[1])
    thechar = parts[1][0]
    passwd = parts[2]
    count = 0
    for i in range(0, len(passwd)):
        if passwd[i] == thechar:
            count = count + 1
    if count >= themin and count <= themax:
        return True
    return False


def process_password2(line):
    line = line.strip()
    parts = line.split(" ")
    minmax = parts[0].split("-")
    themin = int(minmax[0])-1
    themax = int(minmax[1])-1
    thechar = parts[1][0]
    passwd = parts[2]
    a = passwd[themin] == thechar
    b = passwd[themax] == thechar
    return (a and not b) or (not a and b)


filename = "input.txt"
count = 0
with open(filename, "r") as f:
    for line in f.readlines():
        result = process_password2(line)
        if result:
            count = count + 1

print(count)

File: day2/day2.jl

function policy1(line)
    range, ch, passwd = split(line, " ")
    minmax = split(range, "-")
    min = parse(Int64, minmax[1])
    max = parse(Int64, minmax[2])
    thech = ch[1]
    count = 0
    for i in eachindex(passwd)
        if passwd[i] == thech
            count = count + 1
        end
    end
    if count >= min && count <= max 
        return true
    end
    return false
end

function policy2(line)
    range, ch, passwd = split(line, " ")
    minmax = split(range, "-")
    min = parse(Int64, minmax[1])
    max = parse(Int64, minmax[2])
    thech = ch[1]
    count = 0
    xor(thech == passwd[min], thech == passwd[max])
end
open("input.txt", "r") do f
    goodpass = 0
    for line in eachline(f) 
        if policy2(line) 
            goodpass = goodpass + 1
        end
    end
    println("Good Passwords ", goodpass)
end